georest/tests/test_geoserver.py

```diff
@@ -409,16 +471,62 @@ def test_delete_file_from_mosaic(connect_to_gs_catalog):
     cat.delete_granule.assert_not_called()
 
     # This is the structure returned by cat.list_granules()
-    granules = {"features":
-                [{"properties": {"location": "/mnt/data/europe_airmass.tif"},
-                  "id": "file-id"}]
-                }
+    granules = {"features": [{"properties": {"location": "/mnt/data/europe_airmass.tif"}, "id": "file-id"}]}
     cat.list_granules.return_value = granules
 
     delete_file_from_mosaic(config, fname)
     cat.delete_granule.assert_called()
 
 
+@mock.patch("georest.connect_to_gs_catalog")
+def test_delete_file_from_mosaic_layertemplate(connect_to_gs_catalog):
+    """Test deleting files from image mosaic."""
+    from georest import delete_file_from_mosaic
+
+    # This is the structure returned by cat.mosaic_coverages()
+    coverages = {
+        "coverages": {
+            "coverage": [
+                {"name": "airmass1_europe"},
+                {"name": "airmass2_europe"},
+                {"name": "airmass1_global"},
+                {"name": "airmass2_global"},
+            ],
+        }
+    }
+    cat = mock.MagicMock()
+    cat.mosaic_coverages.return_value = coverages
+    connect_to_gs_catalog.return_value = cat
+
+    config = {
+        "workspace": "satellite",
+        "geoserver_target_dir": "/mnt/data",
+        "file_pattern": "{area}_{productname}.tif",
+        "layer_name_template": "{productname}_{area}",
+        "delete_granule_layer_options": {"area": ["europe", "global"], "productname": ["airmass1", "airmass2"]},
+    }
+    fnames = ["europe_airmass1.tif", "global_airmass1.tif", "europe_airmass2.tif", "global_airmass2.tif"]
+
+    for fname in fnames:
+        delete_file_from_mosaic(config, fname)
+
+    connect_to_gs_catalog.assert_called_with(config)
+    assert cat.get_store.call_count == len(fnames)
+    assert "call('airmass1_europe', 'satellite')" in str(cat.get_store.call_args_list)
+    assert "call('airmass2_europe', 'satellite')" in str(cat.get_store.call_args_list)
+    assert "call('airmass1_global', 'satellite')" in str(cat.get_store.call_args_list)
+    assert "call('airmass1_global', 'satellite')" in str(cat.get_store.call_args_list)
```

Comment:
Copy & paste error, I'll fix this.
```suggestion
    assert "call('airmass2_global', 'satellite')" in str(cat.get_store.call_args_list)
```